crates/viewer/re_space_view_dataframe/src/time_range_table.rs

```diff
@@ -0,0 +1,267 @@
+    //
+
+    // TODO(ab): add support for filtering components more narrowly.
+    let sorted_components: BTreeSet<_> = {
+        re_tracing::profile_scope!("query components");
+
+        // Produce a sorted list of all components that are present in one or more entities.
+        // These will be the columns of the table.
+        query
+            .iter_all_data_results()
+            .filter(|data_result| data_result.is_visible(ctx))
+            .flat_map(|data_result| {
+                ctx.recording_store()
+                    .all_components(&query.timeline, &data_result.entity_path)
+                    .unwrap_or_default()
+            })
+            // TODO(#4466): make showing/hiding indicators components an explicit optional
+            .filter(|comp| !comp.is_indicator_component())
+            .collect()
+    };
+
+    //
+    // Build the full list of rows, along with the chunk where the data is. Rows are keyed by an
+    // (entity, time, row_id) tuple (see function docstring). These keys are mapped to the
+    // corresponding chunk that contains the actual data.
+    //
+    // We build a big, monolithic iterator for all the rows. The following code builds that by
+    // breaking it into several functions for sanity purpose, from innermost to outermost.
+    //
+
+    type RowKey = (EntityPath, TimeInt, RowId);
+
+    #[inline]
+    fn map_chunk_indices_to_key_value_iter<'a>(
+        indices_iter: impl Iterator<Item = (TimeInt, RowId)> + 'a,
+        chunk: Arc<Chunk>,
+        entity_path: EntityPath,
+        resolved_time_range: ResolvedTimeRange,
+    ) -> impl Iterator<Item = (RowKey, Arc<Chunk>)> + 'a {
+        indices_iter
+            .filter(move |(time, _)| time.is_static() || resolved_time_range.contains(*time))
+            .map(move |(time, row_id)| {
+                let chunk = chunk.clone();
+                ((entity_path.clone(), time, row_id), chunk)
+            })
+    }
+
+    #[inline]
+    fn entity_components_to_key_value_iter<'a>(
+        ctx: &ViewerContext<'_>,
+        entity_path: &'a EntityPath,
+        component: &'a ComponentName,
+        timeline: Timeline,
+        resolved_time_range: ResolvedTimeRange,
+    ) -> impl Iterator<Item = (RowKey, Arc<Chunk>)> + 'a {
+        let range_query = RangeQuery::new(timeline, resolved_time_range);
+
+        ctx.recording_store()
+            .range_relevant_chunks(&range_query, entity_path, *component)
+            .into_iter()
+            // This does two things:
+            // 1) Filter out instances where `chunk.iter_indices()` returns `None`.
+            // 2) Exploit the fact that the returned iterator (if any) is *not* bound to the
+            //    lifetime of the chunk (it has an internal Arc).
+            .filter_map(move |chunk| {
+                chunk
+                    .clone()
+                    .iter_indices(&timeline)
+                    .map(|iter_indices| (iter_indices, chunk))
```

Comment:
You might want to filter the chunk itself here (`chunk.range(range_query)`).

The real APIs :tm: will do that for you so it's not that important but eh.

Reply:
Good call. Fixed.